### PR TITLE
DJED-164: Extend PAB functionalities for managing local database

### PR DESCRIPTION
**Pull request by j-mueller**
* Use `IORef` instead of `TVar` for `InstancesState` and `InMemInstances`
* Add `DeleteState` to `Contract` effect and call it from `stopInstance`
* Add `NoChainSyncEvents` constructor to `NodeMode`, which disables the chain sync handler that's used in `BlockchainEnv`

TO DO

- [ ] Stop waiting on instance updates after a timeout
  

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Review by etiennejf — COMMENTED** [review]
Plz check my comments.

**Comment by etiennejf** [comment]
I think that we should add a comment in the pab config file w.r.t. NoChainSyncEvents. Moreover? I've checked the code into more details but is there a correlation between the local wallet config and AlonzoMode ?? If so maybe we should also add an extra check to make sure that if NoChainSyncEvents is specified then only the remote wallet config is authorized.

**Comment by j-mueller** [comment]
> is there a correlation between the local wallet config and AlonzoMode 

hm, the only place I could find where there is such as correlation is in [`plutus-pab-executables/pab-cli/App.hs`](https://github.com/etiennejf/plutus-apps/blob/main/plutus-pab-executables/pab-cli/App.hs#L127) which we don't call from Djed

**Review by etiennejf — APPROVED** [review]
The benchmarking performed on this version shows a reduction of more 79% in terms of memory consumption. However, there is no significant win in terms of response time.

File: plutus-pab/src/Plutus/PAB/Core.hs [hunk 1]
```diff
@@ -518,11 +518,13 @@ waitForInstanceState ::
   PABAction t env ContractActivityStatus
 waitForInstanceState extract instanceId = do
   is <- instanceStateInternal instanceId
-  liftIO $ STM.atomically $ do
-    ms <- extract is
-    case ms of
-     Nothing     -> empty
-     Just status -> pure status
+  PABEnvironment{endpointTimeout} <- ask @(PABEnvironment t env)
```
[the remark this answers is not in the document]
Comment (etiennejf) [reply plutus-pab/src/Plutus/PAB/Core.hs]: We now have two timeout values: One when invoking endpoints and one when querying the instance state.

**Pull request merged**